frontend/spec/features/address_spec.rb

```diff
@@ -16,6 +16,98 @@
     @zipcode_label_css = '#b_zipcode_label'
   end
 
+
+  context 'store checkout_zone', js: true do
+    let!(:store) { create(:store, default: true) }
+    let!(:north_america_zone) do
+      usa = create(:country, name: 'United States of America', iso_name: 'UNITED STATES')
+      create(:zone, name: 'North America', kind: 'country', default_tax: true).tap do |zone|
+        zone.members << create(:zone_member, zoneable: usa)
+      end
+    end
+
+    let!(:eu_vat_zone) do
+      denmark = create(:country, name: 'Denmark', iso_name: 'DNK')
+      create(:zone, name: 'EU_VAT', kind: 'country', default_tax: true).tap do |zone|
+        zone.members << create(:zone_member, zoneable: denmark)
+      end
+    end
+
+    context 'when store have checkout_zone_id attribute' do
+      before do
+        store.update!(checkout_zone_id: north_america_zone.id)
+
+        add_to_cart(mug) do
+          click_link 'Checkout'
+        end
+      end
+
+      it 'address form contain selected zone' do
+        expect(page.find('#order_bill_address_attributes_country_id').text).to eq 'United States of America'
+      end
+    end
+
+    context 'user account', js: true do
+      context 'address form in user account' do
+        context 'with checkout_zone' do
+          before do
+            store.update!(checkout_zone_id: north_america_zone.id)
```

Comment:
@przemosk let's test it with some other country/zone please, US is the default country when no zone is set so this could be a false positive

Reply:
@damianlegawiec changed to `Asia` zone